`miri` no longer builds after rust-lang/rust#99553

Hello, this is your friendly neighborhood mergebot.
After merging PR rust-lang/rust#99553, I observed that the tool miri has failing tests.
A follow-up PR to the repository https://github.com/rust-lang/miri is needed to fix the fallout.

cc @ChrisDenton, do you think you would have time to do the follow-up work?
If so, that would be great!

Looks like the `cfg(miri)` codepath doesn't build:
```
error[E0133]: call to unsafe function is unsafe and requires unsafe function or block
    --> /home/r/.rustup/toolchains/miri/lib/rustlib/src/rust/library/std/src/sys/windows/compat.rs:254:21
     |
254  |                       Module::new($module).map(preload);
     |                       ^^^^^^^^^^^^^^^^^^^^ call to unsafe function
     |
    ::: /home/r/.rustup/toolchains/miri/lib/rustlib/src/rust/library/std/src/sys/windows/c.rs:1247:1
     |
1247 | / compat_fn_optional! {
1248 | |     pub static SYNCH_API: &CStr = ansi_str!("api-ms-win-core-synch-l1-2-0");
1249 | |
1250 | |     // >= Windows 8 / Server 2012
...    |
1258 | |     pub fn WakeByAddressSingle(Address: LPVOID) -> ();
1259 | | }
     | |_- in this macro invocation
```

--- a/library/std/src/sys/windows/compat.rs
+++ b/library/std/src/sys/windows/compat.rs
@@ -251,7 +251,7 @@ macro_rules! compat_fn_optional {
             pub fn option() -> Option<F> {
                 let mut func = NonNull::new(PTR.load(Ordering::Relaxed));
                 if func.is_none() {
-                    Module::new($module).map(preload);
+                    unsafe { Module::new($module).map(preload) };
                     func = NonNull::new(PTR.load(Ordering::Relaxed));
                 }
                 unsafe {

--- a/library/std/src/sys/windows/compat.rs
+++ b/library/std/src/sys/windows/compat.rs
@@ -180,8 +180,8 @@ macro_rules! compat_fn_with_fallback {
 
             fn load_from_module(module: Option<Module>) -> F {
                 unsafe {
-                    static symbol_name: &CStr = ansi_str!(sym $symbol);
-                    if let Some(f) = module.and_then(|m| m.proc_address(symbol_name)) {
+                    static SYMBOL_NAME: &CStr = ansi_str!(sym $symbol);
+                    if let Some(f) = module.and_then(|m| m.proc_address(SYMBOL_NAME)) {
                         PTR.store(f.as_ptr(), Ordering::Relaxed);
                         mem::transmute(f)
                     } else {
@@ -262,8 +262,8 @@ macro_rules! compat_fn_optional {
             #[allow(unused)]
             pub(in crate::sys) fn preload(module: Module) {
                 unsafe {
-                    let symbol_name = ansi_str!(sym $symbol);
-                    if let Some(f) = module.proc_address(symbol_name) {
+                    static SYMBOL_NAME: &CStr = ansi_str!(sym $symbol);
+                    if let Some(f) = module.proc_address(SYMBOL_NAME) {
                         PTR.store(f.as_ptr(), Ordering::Relaxed);
                     }
                 }

